spec/uber_task/uber_task_spec.rb

```diff
@@ -0,0 +1,58 @@
+# frozen_string_literal: true
+
+describe UberTask do
+  describe '.on_report' do
+    context 'when no block is passed' do
+      it 'returns nil' do
+        described_class.run do |task|
+          expect(described_class.on_report).to be_nil
+        end
+      end
+    end
+
+    context 'when a block is passed' do
+      it 'returns nil' do
+        described_class.run do |task|
+          result = described_class.on_report do
+            "Added some logs"
+          end
+
+          expect(result.call).to eq("Added some logs")
+        end
+      end
+    end
+  end
+  
+  describe '.on_retry' do
+    context 'when no block is passed' do
+      it 'returns the wait time and set handlers retry to nil' do
+        described_class.run do |task|
+          described_class.on_retry(report: false, wait: 12)
+
+          expect(task.handlers[:retry]).to be_nil
+          expect(task.retry_info.report).to be false
+          expect(task.retry_info.wait).to eq(12)
```

Comment:
We also need to update the `on_retry` method to always return nil. The returned value was not intentional.